### ISSUE TITLE
Feature request: dots every 1 second

### ISSUE THREAD

**Issue by ariel-phet** [phetsims/projectile-motion#152]
@arouinfar @andrea-phet Mike Dubson told @kathy-phet he really liked the new dots, but missed having a dot every one second. Obviously, there is a dot every 1 second but it does not "stand out" currently. 

It seems we could accomplish this feature by potentially having the dot be noticeably larger every one second, or some other way to make it stand out (perhaps some sort of stroke). Basically have the dots feel like minor and major tick marks of a ruler.

**Comment by ariel-phet** [comment]
@arouinfar might be worth mocking up a few options to see if this would work

**Comment by arouinfar** [comment]
Sounds good @ariel-phet. I really like the idea to map the dots to minor/major tick marks. I'll mock up a few options.

**Comment by arouinfar** [comment]
Timing dot parameters:
~3.5 pt in diameter
50% opacity (max)
Black stroke with no fill

The dots are small and transparent to improve contrast with the projectile, and I believe the lack of a fill was done for performance reasons. Since there are fewer 1s markers, I'm hoping that the performance impact of a fill would be negligible.

A few options:

(1) Large black dots, max opacity 100%
![image](https://user-images.githubusercontent.com/8419308/30608933-208ada44-9d37-11e7-980e-2079ef43965b.png)

(2) Large black dots, opacity matches regular timing dots
![image](https://user-images.githubusercontent.com/8419308/30608970-3f860b80-9d37-11e7-9981-8709a41f435e.png)

(3) Yellow dots similar to apex dot
![image](https://user-images.githubusercontent.com/8419308/30609001-57e60ee6-9d37-11e7-83b2-c689dbd20cb1.png)

(4) Black crosshairs (similar to Flash)
![image](https://user-images.githubusercontent.com/8419308/30609027-71d69898-9d37-11e7-92e2-f000f994c7b3.png)

(1) and (2) are have a major/minor tick mark feel in comparison to the regular timing dots. These dots might be a bit hard to distinguish from the cannonball, but that may not be an issue because the dots are relatively infrequent. I have a slight preference for (1) because the 1s dots are a bit easier to spot.

(3) doesn't have the same major/minor relationship between the 1s  and 0.1s  dots, but I think the yellow contrasts really nicely against the path. I recommend you open this screenshot in another window to review as the dots aren't quite as visible when the window is small.

(4) is a nod to the Flash markers, but I prefer the cleaner look of the circular markers.

**Comment by arouinfar** [comment]
Can you take a look @ariel-phet?

**Comment by phet-steele** [comment]
Which has greater "tracer tool priority" when close to each other: the apex point or 1 second point? If they are close and overlapping, will we want to see both or will one obscure the other? Which one will the tracer tool prefer to measure? We already have special code to increase priority of the apex point over all other points so its easy to measure in a cluster of points (see #138).

**Comment by arouinfar** [comment]
Great question @phet-steele. I think we'd continue to prioritize the apex dot, as it's more pedagogically interesting. The 1s dots are still timing dots, after all, but with improved visibility.

**Comment by ariel-phet** [comment]
@arouinfar I think I like the yellow dots the best. Especially with the cannonball type projectile my main worry  was that larger dots could be confusing. The different color dot seems to be a clean solution.

Also agreed to prioritize the apex dot.

**Comment by phet-steele** [comment]
@ariel-phet @arouinfar sorry to butt in again, but yellow dots are tricky when combined with the yellow halo. Maybe a special halo color for this dot? Or a different color dot all together? Color options that were previously deliberated for something else (for use with either the dot or halo):

* White
* Gray -- likely would have contrast issues against the pedestal
* Cyan -- nice and bright, but likely has poor contrast against the sky
* Hunter/Safety orange -- could work, but isn’t pretty (2nd choice after white)

**Comment by kathy-phet** [comment]
I kind of like the simplicity of the black larger dots, and since they are stationary I don't see that they would be confused with the projectile. However, its just a preference. I can be OK with other options too. Even the crosses.

**Comment by arouinfar** [comment]
I prefer option (3) with the yellow dots. I think they have nice contrast against the other colors in the play area and won't be confused for a cannonball. 

@phet-steele no need to be sorry, we should be cognizant of halo color too. Personally, I don't think the yellow halo is particularly problematic here (apex dot has green halo) but we will certainly think about it when we discuss at design meeting.

**Comment by arouinfar** [comment]
Here's a mockup with the 1s markers with at ~2x the radius of the 0.1s markers. I changed `DOT_RADIUS` from 1.65 to 3. What do you think @ariel-phet @kathy-phet?
![screen shot 2017-10-05 at 5 24 53 pm](https://user-images.githubusercontent.com/8419308/31256888-656f0a78-a9f2-11e7-95d4-80e41938e0d6.png)

**Comment by ariel-phet** [comment]
I really like this solution.

**Comment by jbphet** [comment]
This is implemented and available here https://www.colorado.edu/physics/phet/dev/html/projectile-motion/1.0.2-rc.1/projectile-motion_en.html.

@arouinfar - please test this out and give final approval before we go live with this.  Feel free to bring in @ariel-phet if you like.

**Comment by arouinfar** [comment]
Thanks @jbphet! I really like the 1s dots. Looks like the mockup, so I don't think @ariel-phet needs to verify too.

**Comment by phet-steele** [comment]
Reopening not because it's broken, but because I question the implementation (the dots look great and function well!).

@jbphet in `ProjectileMotionConstants`, a couple things were added, namely:

```js
var TIME_PER_MINOR_DOT = 100; // milliseconds

var ProjectileMotionConstants = {

TIME_PER_MINOR_DOT: TIME_PER_MINOR_DOT, // milliseconds
TIME_PER_MAJOR_DOT: TIME_PER_MINOR_DOT * 10 // milliseconds, must be a multiple of the other dot size
}
```
Since "major" dots are intended to appear every 1 second, wouldn't `TIME_PER_MAJOR_DOT` break if `TIME_PER_MINOR_DOT` were anything other than 100? Note the multiplication times 10.

If at some point in the future we wanted "minor" dots to show up every half second, we would have to change `TIME_PER_MINOR_DOT` to be equal to 500. This  means `TIME_PER_MAJOR_DOT` would evaluate to 5000, so each major dot would show up every 5 seconds, not 1 second. This would require a second change to have `TIME_PER_MINOR_DOT` multiplied by 2, instead of 10, to get the correct time for major dots.

My feeling is that, if we want to retain the 1 second rule for major dots independently of how often we show minor dots, `TIME_PER_MAJOR_DOT` should be calculated like:

```js
TIME_PER_MAJOR_DOT: (1000 / TIME_PER_MINOR_DOT ) * TIME_PER_MINOR_DOT
```

But that's really really weird and overly complicated, so it would probably be way simpler to straight up define major dots as its own constant:

```js
var TIME_PER_MINOR_DOT = 100; // milliseconds
var TIME_PER_MAJOR_DOT = 1000; // milliseconds, intended to always be one second

var ProjectileMotionConstants = {

TIME_PER_MINOR_DOT: TIME_PER_MINOR_DOT, // milliseconds
TIME_PER_MAJOR_DOT: TIME_PER_MAJOR_DOT // milliseconds
}
```

@jbphet what do you think? If I'm wrong about this, it may be because of the part that states `TIME_PER_MAJOR_DOT` "must be a multiple of the other dot size" (in the code comment) due to some restriction placed elsewhere in the sim. 

@arouinfar am I correct in assuming we would want the major dot intervals to be independent of the minor dot intervals AND that for now they should only ever appear after 1 second?

**Comment by arouinfar** [comment]
> @arouinfar am I correct in assuming we would want the major dot intervals to be independent of the minor dot intervals AND that for now they should only ever appear after 1 second?

Correct, major dots will only appear at 1s intervals, and should be independent of the minor dot interval.

**Comment by phet-steele** [comment]
In `TrajectoyNode.js`, could just change line 105 to this to eliminate the dependency of these being multiples of each other.
```js 
if ( addedPointTimeInMs % TIME_PER_MINOR_DOT === 0 ||
     addedPointTimeInMs % TIME_PER_MAJOR_DOT === 0 ) { ... }
```

**Comment by jbphet** [comment]
I made the large dot spacing independent of the small dot spacing.  The commit is 9fa52a8831127cea2ec6c336867d9d5a1fbae390 (I forgot to reference this issue in the commit message).  Here is a screenshot where I experimented with small dot spacing that is not a multiple of large dot spacing just to verify that it works:

![pm-dots](https://user-images.githubusercontent.com/3082831/32339974-ef98349a-bfbe-11e7-9572-7a77b1393437.PNG)

**Comment by jbphet** [comment]
@phet-steele - the change to make the small and large dot spacing independent of one another has been committed to master and the 1.0 branch.  Please let me know if we are now approved by QA to publish.

**Comment by phet-steele** [comment]
@jbphet please proceed with publishing!

**Comment by jbphet** [comment]
Published in v1.0.2, closing.
